### PR TITLE
Add _no_history support for deleteByCriteria

### PR DESCRIPTION
**Pull request by amma35**
Add _no_history support for deleteByCriteria

### CODE REVIEW

File: inc/commondbtm.class.php [hunk 1]
```diff
@@ -1357,7 +1357,9 @@ function delete(array $input, $force=0, $history=1) {
       }
 
       if ($this->pre_deleteItem()) {
-
+         if (!isset($this->input['_no_history'])) {
```
Comment (orthagh) [inline inc/commondbtm.class.php]: This lines could be upper in the code, no ?

File: inc/commondbtm.class.php [hunk 2]
```diff
@@ -3765,8 +3767,13 @@ function deleteByCriteria($crit=array(), $force=0) {
       if (is_array($crit) && (count($crit) > 0)) {
          $crit['FIELDS'] = 'id';
          $ok = true;
+         $history = 1;
+         if(isset($crit['_no_history'])){
```
Comment (orthagh) [inline inc/commondbtm.class.php]: instead of a crit (which don't corresponds to your need, i'll prefer a new function parameter (set to true at default).

Comment (amma35) [inline inc/commondbtm.class.php]: Ok, I update

File: inc/commondbtm.class.php (hunk 1 above)
Comment (orthagh) [inline inc/commondbtm.class.php]: yes

File: inc/commondbtm.class.php [hunk 3]
```diff
@@ -1343,6 +1343,11 @@ function delete(array $input, $force=0, $history=1) {
          $this->input['_delete'] = $this->input['delete'];
          unset($this->input['delete']);
       }
+
+      if (!isset($this->input['_no_history'])) {
+         $this->input['_no_history'] = $history;
```
Comment (orthagh) [inline inc/commondbtm.class.php]: Last before merge, this part shouldn't be: 
```php
$this->input['_no_history'] = !$history;
```

Comment (amma35) [inline inc/commondbtm.class.php]: Thanks

**Comment by orthagh** [comment]
Last thing, and this will be merged (after travis check)
Could you think about a dedicated unit test ?

**Comment by amma35** [comment]
I add a unit test in PrinterTest for example?

**Comment by trasher** [comment]
Yes, I think it should be enough ;)

**Comment by orthagh** [comment]
very good !

**Comment by trasher** [comment]
Great, thank you!

**Comment by trasher** [comment]
Manually merged into a1f84208c4b9da61925371d1c62674cbd2b14c89

**Pull request closed without merging**